compiler/src/yul/mappers/expressions.rs

```diff
@@ -272,6 +272,25 @@ fn expr_attribute_self(
     }
 }
 
+fn expr_ternary(
+    context: &Context,
+    exp: &Spanned<fe::Expr>,
+) -> Result<yul::Expression, CompileError> {
+    if let fe::Expr::Ternary {
+        if_expr,
+        test,
+        else_expr,
+    } = &exp.node
+    {
+        let yul_test_expr = expr_comp_operation(context, test)?;
```

Comment:
The test needs to use `expr` as well. Otherwise cases like these would fail:

Boolean literal:

```
    pub def bar(input: u256) -> u256:
        return 1 if true else 0
```

Function call:

Boolean literal:

```
    pub def bar(input: u256) -> u256:
        return 1 if self.check(input) else 0
```

Reply:
Thanks for the pointer. I fixed it.